data_structures/queue_using_array.cpp

```diff
@@ -1,69 +1,69 @@
 /*
     Write a program to implement Linear Queue using array.
-
     Functions to implement
         Enqueue (Insertion)
         Dequeue (Deletion)
-
 */
 #include <iostream>
+#include <vector>
 
-#define MAXSIZE 10
+constexpr int max_size = 10;
 
 class Queue_Array {
-    int front;
-    int rear;
-    int size;
-
- public:
-    Queue_Array() {
-        front = -1;
-        rear = -1;
-        size = MAXSIZE;
-    }
-    int *arr = new int[size];
+public:
+    Queue_Array();
     void enqueue(int);
     int dequeue();
-    void display();
+    void display() const;
+private:
+    int front;
```

Comment:
It is recommended to use `uint64_t` for non-negative values or `int64_t` for negative values (or their appropriate size). Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)).

Reply:
Thank you,
What will happen if we use larger size for most cases (`uint64_t`), Will it waste memory?

Reply:
Yes, wasting memory is not good. It could lead to memory overflow, too.